### PR TITLE
Add new temperature control in climate more info

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

In this PR :
- [x] select temperature using slider
- [x] select temperature using buttons
- [x] display current temperature
- [x] display current humidity
- [x] add gallery

Not in this PR (will follow with other PRs) : 
- [ ] change hvac mode
- [ ] change preset mode
- [ ] change swing mode
- [ ] change fan mode
- [ ] select humidity using slider

### Demo

https://github.com/home-assistant/frontend/assets/5878303/5a508f4d-5877-4f32-87d4-616ee27a464b

### Examples

CleanShot 2023-08-07 at 13 03 09

CleanShot 2023-08-07 at 13 03 04

CleanShot 2023-08-07 at 13 02 36

CleanShot 2023-08-07 at 13 03 04

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+export class HaMoreInfoClimateTemperature extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj!: ClimateEntity;
+
+  @state() private _targetTemperature: Partial<Record<Target, number>> = {};
+
+  @state() private _selectTargetTemperature: Target = "low";
+
+  protected updated(changedProp: Map<string | number | symbol, unknown>): void {
```
Comment (bramkragten) [inline src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts:35]: This should be done in `willUpdate`

**Comment by spacegaier** [comment]
I think having the decimal place in the superscript/elevated position (where the unit is) is confusing and looks weird. It is not part of the unit, so it should follow the font size of the value (which it is part of logically).

**Comment by piitaya** [comment]
Feature postponed to next release because UX.UI needs to be discussed

**Comment by piitaya** [comment]
@matthiasdebaat I updated the design according to new figma prototype. 
Only things missing are : 
- the current temperature label (not sure it's needed). I can be nice to test this design with mouse/touch interaction.
- text colors need to be adjusted but we need to figure how we adapt the colors according to [state colors](https://www.home-assistant.io/integrations/frontend/#state-color).

These detail can be done in future PR. This one can be reviewed and we can iterate with future ones because this PR is already big.

File: src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+    this._targetTemperature = {
+      ...this._targetTemperature,
+      [target]: value,
+    };
+    this._selectTargetTemperature = target as Target;
+  }
+
+  private _debouncedCallService = debounce(
+    (target: Target) => this._callService(target),
+    2000
```
Comment (bramkragten) [inline src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts:94]: 2 seconds seems long, wouldn't 1 sec be enough?

File: src/dialogs/more-info/controls/more-info-climate.ts [hunk 3]
```diff
@@ -73,13 +76,60 @@ class MoreInfoClimate extends LitElement {
...
-      stateObj.attributes.target_temp_step ||
-      (hass.config.unit_system.temperature.indexOf("F") === -1 ? 0.5 : 1);
+    const currentTemperature = this.stateObj.attributes.current_temperature;
+    const currentHumidity = this.stateObj.attributes.current_humidity;
 
     const rtlDirection = computeRTLDirection(hass);
 
     return html`
+      ${currentTemperature || currentHumidity
+        ? html` <div class="current">
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-climate.ts:86]: nagging
```suggestion
        ? html`<div class="current">
```

File: src/panels/lovelace/cards/hui-thermostat-card.ts [hunk 4]
```diff
@@ -47,6 +47,7 @@ import { createEntityNotFoundWarning } from "../components/hui-warning";
 import { LovelaceCard, LovelaceCardEditor } from "../types";
 import { ThermostatCardConfig } from "./types";
 
+// Need to align these icon to more info icons
```
Comment (bramkragten) [inline src/panels/lovelace/cards/hui-thermostat-card.ts:50]:
```suggestion
// TODO: Need to align these icon to more info icons
```

**Review by bramkragten — DISMISSED** [review]

File: src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts (hunk 2 above)
Comment (piitaya) [reply src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts:94]: I reused the duration from the old control but we can reduce it 🙂

Comment (piitaya) [inline src/dialogs/more-info/components/climate/ha-more-info-climate-temperature.ts:94]:
```suggestion
    1000
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**